crates/gen-js/src/lib.rs

```diff
@@ -2470,3 +2488,18 @@ impl Source {
         self.ts.push_str(s);
     }
 }
+
+fn is_nullable(iface: &Interface, ty: &Type) -> bool {
```

Comment:
Could this perhaps return `Option<&Type>` so the match here doesn't have to be redone above?

Reply:
Also, would it be possible to remove this method and use `self.is_nullable_option` instead? (or some sort of shared helper method)

Reply:
Before I blindly copied and didn't read too much into the details, but now I'm a little confused.

```rust
match variant.as_option() {
        Some(ty) => match ty {
              Type::Id(id) => match &iface.types[*id].kind {
                  TypeDefKind::Variant(v) => !self.is_nullable_option(iface, v),
                  ty => true,
              },
              _ => true,
          },
          None => false,
      }
```

Why is this not just `variant.as_option().is_some()`?  Wouldn't anything with `option<T>` be a nullable option, regardless of `T`?

Reply:
Nvm I see.  `option<option<T>>` becomes `Option<T | null>` in TS.